projects/project-2/readings/Project 2 - SPARQL library

```diff
@@ -0,0 +1,297 @@
+Identifying Potentially Overspecified Class Definitions (level 5 - 5 points)
+Detecting Equivalent Properties with Different Ranges  (level 5 - 5 points)
+Analyzing Class Hierarchies for Inconsistent Type Definitions (level 5 - 5 points)
+Detecting Potentially Misplaced Inverse Object Properties (level 5 - 5 points)
+Identifying Classes with No Properties (Level 4 - 10 points)
+Identifying Classes Without Labels (Level 4 - 10 points)
+Query to Detect Missing Annotations (Labels or Comments) Level 4 - 10 points
+Detecting Properties Shared Across Multiple CCO Classes Level 3 - 20 points
+Detecting Inconsistent Property Ranges Across Subclass Hierarchies Level 3 - 20 points
+
+1. Detect Classes Without Any Instances (Level 8 - 0 points)
+
+Title: Empty Class Detection
+Constraint Description: Identifies classes that have no instances associated with them.
+Severity: Warning
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?class
+WHERE {
+  ?class a rdfs:Class .
+  FILTER NOT EXISTS {
+    ?instance a ?class .
+  }
+}
+
+Purpose: ​This query retrieves all classes that do not have any instances associated with them.​ 
+It uses FILTER NOT EXISTS to ensure that only classes without any defined instances are selected,helping to identify potentially unused or unnecessary classes in the ontology.
+
+2. Identifying Disjoint Classes with Common Instances
+
+Title: Disjoint Class Violation
+Constraint Description: Detects instances that belong to classes declared as disjoint.
+Severity: Error
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+SELECT ?class1 ?class2 ?instance
+WHERE {
+  ?instance a ?class1 .
+  ?instance a ?class2 .
+  ?class1 owl:disjointWith ?class2 .
+}
+
+The primary intention of this query is to identify pairs of classes that are designated as disjoint but have instances that belong to both classes.​ 
+This situation indicates a potential inconsistency within the ontology since disjoint classes should not share any instances.
+
+Running this query enables to detect potential issues in the class hierarchy and relationships, ensuring the integrity of the ontology is maintained. 
+
+3. Detecting Unused Object Properties (level 7 - 2 points)
+
+Title: Unused Object Property Detection
+Constraint Description: Identifies object properties that are defined but not used in any instances.
+Severity: Warning
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+SELECT ?property
+WHERE {
+  ?property a owl:ObjectProperty .
+  FILTER NOT EXISTS {
+    ?subject ?property ?object .
+  }
+}
+
+​The aim of this query is to identify object properties within the ontology that are defined but not utilized in any instances.​ 
+Detecting such unused properties helps maintain a clean and efficient ontology by identifying potentially redundant definitions.
+
```

Comment:
Do you have to remove all the unused properties? They could be placeholders for data that hasn't been added yet. Some properties may be intended for future use or may be waiting for data to be ingested. But if CCO is complete on its own, then I see that this query will be valuable.

Reply:
I´ll check this!